src/rules/unnecessaryConstructorRule.ts

```diff
@@ -45,6 +45,15 @@ export class Rule extends Lint.Rules.AbstractRule {
 const isEmptyConstructor = (node: ts.ConstructorDeclaration): boolean =>
     node.body !== undefined && node.body.statements.length === 0;
 
+const isAccessRestrictingConstructor = (
+    node: ts.ConstructorDeclaration,
+    context: Lint.WalkContext,
+): boolean =>
+    // No modifiers implies public
+    node.modifiers != undefined &&
+    // If this has any modifier that isn't public, it's doing something
+    node.modifiers.findIndex(modifier => modifier.getText(context.sourceFile) !== "public") >= 0;
```

Comment:
It's a bit speedier to check the modifier's `.kind`, which might be of type `ts.SyntaxKind.PublicKeyword`. 

```suggestion
    node.modifiers.some(modifier => modifier.kind !== ts.SyntaxKind.PublicKeyword);
```

Reply:
Nice – will make this update! Thanks :)